pkg/epp/datalayer/metrics/factories.go

```diff
@@ -0,0 +1,184 @@
+/*
+Copyright 2025 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package metrics
+
+import (
+	"encoding/json"
+	"flag"
+	"fmt"
+	"strconv"
+
+	"sigs.k8s.io/gateway-api-inference-extension/pkg/epp/plugins"
+)
+
+const (
+	DataSourceType = "metrics-data-source"
+	ExtractorType  = "model-server-protocol-metrics"
+)
+
+// Configuration parameters for metrics data source and extractor.
+type (
+	// Data source configuration parameters
+	datasourceParams struct {
+		// Scheme defines the protocol scheme used in metrics retrieval (e.g., "http").
+		Scheme string // `json:"scheme"`
+		// Path defines the URL path used in metrics retrieval (e.g., "/metrics").
+		Path string // `json:"path"`
+		// InsecureSkipVerify defines whether model server certificate should be verified or not.
+		InsecureSkipVerify bool // `json:"insecureSkipVerify"`
+	}
+
+	// Extractor configuration parameters
+	extractorParams struct {
+		// QueueRequestsSpec defines the metric specification string for retrieving queued request count.
+		QueueRequestsSpec string // `json:"queuedRequestsSpec"`
+		// RunningRequestsSpec defines the metric specification string for retrieving running requests count.
+		RunningRequestsSpec string // `json:"runningRequestsSpec"`
+		// KVUsage defines the metric specification string for retrieving KV cache usage.
+		KVUsageSpec string // `json:"kvUsageSpec"`
+		// LoRASpec defines the metric specification string for retrieving LoRA availability.
+		LoRASpec string // `json:"loraSpec"`
+		// CacheInfoSpec defines the metrics specification string for retrieving KV cache configuration.
+		CacheInfoSpec string // `json:"cacheInfoSpec"`
+	}
+)
+
+// DataSourceFactory is a factory function used to instantiate data layer's
+// metrics data source plugins specified in a configuration.
+func DataSourceFactory(name string, parameters json.RawMessage, handle plugins.Handle) (plugins.Plugin, error) {
```

Comment:
This name is too generic

```suggestion
func MetricsDataSourceFactory(name string, parameters json.RawMessage, handle plugins.Handle) (plugins.Plugin, error) {
```